tests/initcoder_test.go

```diff
@@ -0,0 +1,156 @@
+		// createa a transaction with 1 word of code.
+		receipt1, err := runTransactionWithCodeSizeAndGas(require, net, 1, sufficientGas)
+		require.NoError(err)
+		require.Equal(types.ReceiptStatusSuccessful, receipt1.Status, "failed on transfer to empty receiver with valid code")
+
+		// createa a transaction with 2 byte of code.
+		receipt2, err := runTransactionWithCodeSizeAndGas(require, net, 2, sufficientGas)
+		require.NoError(err)
+		require.Equal(types.ReceiptStatusSuccessful, receipt2.Status, "failed on transfer to empty receiver with valid code")
+
+		difference := receipt2.GasUsed - receipt1.GasUsed
+		require.Equal(difference, zeroByteCost, "gas difference between 1 and 2 words should be 4, instead got %d",
+			difference)
+	})
+
+	t.Run("make a transaction without enough gas for init code size", func(t *testing.T) {
+		// as specified in https://eips.ethereum.org/EIPS/eip-3860#rules,
+		// this is similar to transactions considered invalid for not meeting the intrinsic gas cost requirement.
+		_, err := runTransactionWithCodeSizeAndGas(require, net, MAX_INIT_CODE_SIZE+1, sufficientGas)
+		require.ErrorContains(err, "intrinsic gas too low", "unexpectedly succeeded to create contract with init code length greater than MAX_INITCODE_SIZE")
+	})
+}
+
+func testForVariant(t *testing.T, net *IntegrationTestNet,
+	contract *initcode.Initcode, receipt *types.Receipt, variant variant,
+	gasForContract, wordCost uint64) {
+	require := require.New(t)
+
+	var gasCostFor2Words uint64 = wordCost * 2
+	// we use enough gas for all tests to afford cost of 3 words as well.
+	var gasFor3Words = gasForContract + gasCostFor2Words
+	t.Run("create transaction with enough gas for init code", func(t *testing.T) {
+		// we need to provide same gas to all transactions so that we can compare the cost.
+		cost1Word, _ := createAndCheckContractWithCodeLenAndGas(require, net, contract, variant, 30, gasFor3Words)
+		cost2Words, _ := createAndCheckContractWithCodeLenAndGas(require, net, contract, variant, 42, gasFor3Words)
+		cost3Words, _ := createAndCheckContractWithCodeLenAndGas(require, net, contract, variant, 90, gasFor3Words)
+		require.Equal(cost2Words-cost1Word, wordCost,
+			"cost difference between 1 and 2 words should be %v, instead got %d", wordCost, cost2Words-cost1Word)
+		require.Equal(cost3Words-cost1Word, gasCostFor2Words,
+			"cost difference between 1 and 3 words should be %v, instead got %d", gasCostFor2Words, cost3Words-cost1Word)
+	})
+
+	t.Run("create transaction without enough gas for init code", func(t *testing.T) {
+		receipt = createContractWithCodeLenAndGas(require, net, variant, 30, gasForContract-wordCost)
+		require.Equal(types.ReceiptStatusFailed, receipt.Status,
+			"unexpectedly succeeded to create contract without enough gas")
+	})
+
+	t.Run("create transaction with max init code size", func(t *testing.T) {
+		_, _ = createAndCheckContractWithCodeLenAndGas(require, net, contract, variant, MAX_INIT_CODE_SIZE, sufficientGas)
```

Comment:
both returns are numerical values that are unrelated for this tests case. in this case we just want to check that it can run with those parameters and no error occurred. Errors are checked inside the function.

Reply:
errcheck will only report non-catched errors, other ignored return values can be ignored.